src/java/org/apache/cassandra/tcm/ClusterMetadataService.java

```diff
@@ -844,11 +844,23 @@ private Pair<State, Processor> delegateInternal()
         @Override
         public Commit.Result commit(Entry.Id entryId, Transformation transform, Epoch lastKnown, Retry.Deadline retryPolicy)
         {
-            Pair<State, Processor> delegate = delegateInternal();
-            Commit.Result result = delegate.right.commit(entryId, transform, lastKnown, retryPolicy);
-            if (delegate.left == LOCAL || delegate.left == RESET)
-                replicator.send(result, null);
-            return result;
+            while (!retryPolicy.reachedMax())
+            {
+                try
+                {
+                    Pair<State, Processor> delegate = delegateInternal();
+                    Commit.Result result = delegate.right.commit(entryId, transform, lastKnown, retryPolicy);
+                    ClusterMetadataService.State state = delegate.left;
+                    if (state == LOCAL || state == RESET)
+                        replicator.send(result, null);
+                    return result;
+                }
+                catch (NotCMSException e)
+                {
+                    retryPolicy.maybeSleep();
+                }
+            }
+            return Commit.Result.failed(ExceptionCode.SERVER_ERROR, "Could not commit " + transform.kind() + " at epoch " + lastKnown);
```

Comment:
@ifesdjeen you might want to look here.  You provided the following feedback in https://github.com/apache/cassandra/pull/3486

> nit: should we use string format here?
https://github.com/apache/cassandra/pull/3486#discussion_r1732676610